src/lib.rs

```diff
@@ -19,7 +18,7 @@ use napi::Status::GenericFailure;
 use napi::{self, Env};
 use nix::errno::Errno;
 use nix::fcntl::{fcntl, FcntlArg, FdFlag, OFlag};
-use nix::poll::{poll, PollFd, PollFlags, PollTimeout};
+use nix::libc::{FIONREAD, TIOCOUTQ, ioctl};
```

Comment:
any chance we can use [`nix::sys::ioctl`](https://docs.rs/nix/latest/nix/sys/ioctl/) instead of `nix::libc::ioctl`? they're less unsafe because they give you the correct types and stuff.